From b80a31f2d8debd7681d141b03ce167701549be4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=8D=C3=B1igo=20Huguet?= <ihuguet@redhat.com>
Date: Fri, 21 Jan 2022 08:49:03 +0100
Subject: [PATCH 25/36] rtw89: don't kick off TX DMA if failed to write skb
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Bugzilla: http://bugzilla.redhat.com/2033291

commit a58fdb7c843a37d6598204c6513961feefdadc6a
Author: Ping-Ke Shih <pkshih@realtek.com>
Date:   Wed Dec 1 17:38:16 2021 +0800

    rtw89: don't kick off TX DMA if failed to write skb

    This is found by Smatch static checker warning:
            drivers/net/wireless/realtek/rtw89/mac80211.c:31 rtw89_ops_tx()
            error: uninitialized symbol 'qsel'.

    The warning is because 'qsel' isn't filled by rtw89_core_tx_write() due to
    failed to write. The way to fix it is to avoid kicking off TX DMA, so add
    'return' to the failure case.

    Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Ping-Ke Shih <pkshih@realtek.com>
    Signed-off-by: Kalle Valo <kvalo@kernel.org>
    Link: https://lore.kernel.org/r/20211201093816.13806-1-pkshih@realtek.com

Signed-off-by: Íñigo Huguet <ihuguet@redhat.com>
---
 drivers/net/wireless/realtek/rtw89/mac80211.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/realtek/rtw89/mac80211.c b/drivers/net/wireless/realtek/rtw89/mac80211.c
index 757685de6b07..a322259f4cc4 100644
--- a/drivers/net/wireless/realtek/rtw89/mac80211.c
+++ b/drivers/net/wireless/realtek/rtw89/mac80211.c
@@ -27,6 +27,7 @@ static void rtw89_ops_tx(struct ieee80211_hw *hw,
 	if (ret) {
 		rtw89_err(rtwdev, "failed to transmit skb: %d\n", ret);
 		ieee80211_free_txskb(hw, skb);
+		return;
 	}
 	rtw89_core_tx_kick_off(rtwdev, qsel);
 }
-- 
2.13.6

